dbscan/internal_test.go

```diff
@@ -40,3 +42,41 @@ func DoTestRowScannerStartCalledExactlyOnce(t *testing.T, queryRows queryRowsFun
 
 	mockStart.AssertNumberOfCalls(t, "Execute", 1)
 }
+
+type Level1 struct {
+	Level2
+	Foo string
+	Sub Sub
+}
+
+type Sub struct {
+	Foo string
+}
+
+type Level2 struct {
+	Level3
+}
+
+type Level3 struct {
+	Level4
+}
+
+type Level4 struct {
+	Zero  string
+	One   string
+	Two   string `db:"-"`
+	Three string `db:"not_three"`
+}
+
+func TestFieldIndex(t *testing.T) {
+	s := Level1{}
+	m := getColumnToFieldIndexMap(reflect.TypeOf(s))
+
+	assert.Equal(t, []int{0, 0, 0, 0}, m["zero"])
+	assert.Equal(t, []int{0, 0, 0, 1}, m["one"])
+	assert.Equal(t, []int{0, 0, 0, 3}, m["not_three"])
+	assert.Equal(t, []int{1}, m["foo"])
+	_, found := m["two"]
+	assert.Falsef(t, found, "column two should be absent")
+	assert.Equal(t, []int{2, 0}, m["sub.foo"])
+}
```

Comment:
I think we can cover that case by testing the exported method: `RowScanner.Scan()` as all other tests in `rowscanner_test.go`. 
Because without your fix the scan would be incorrect and the field values would be overridden.

Reply:
sure. This was a quick reproduction that turned into a test but, yes, your original test needs to be augmented instead (and had I spent a bit more time exploring the code base I would have done that..). If that test explores one more nesting deeper it should trip over the issue.

Reply:
(two more, as it turns out, cap 1, 2, 4 means level 3 works (len 2 to len 3), level 4 doesn't)